File: python/day-2/a.py
def main():
    instructions = open('in.txt').read().splitlines()
    horizontal = 0
    depth = 0
    for inst in instructions:
        direction, quantity = inst.split()
        quantity = int(quantity)

        if direction == "forward":
            horizontal += quantity
        elif direction == "down" :
            depth += quantity
        elif direction == "up":
            depth -= quantity
        
    print(horizontal * depth)

main()

File: rust/day-2/a.rs
use std::fs;
use std::error::Error;

fn main() -> Result<(), Box<dyn Error>> {
    let inp = fs::read_to_string("in.txt")?;

    let instructions: Vec<&str> = inp
        .lines()
        .collect();

    let mut horizontal: usize = 0;
    let mut depth: usize = 0;

    for inst in instructions {
        let pair: Vec<&str> = inst.split_whitespace().collect();
        let direction = pair[0];
        let quantity: usize = pair[1].parse().unwrap();

        if direction == "forward" {
            horizontal += quantity
        } else if direction == "down" {
            depth += quantity
        } else if direction == "up" {
            depth -= quantity
        }
    }

    println!("{}", horizontal * depth);
    Ok(())
}